graphs/eacs/src/test/kotlin/com/anaplan/engineering/azuki/graphs/analysis/directed/HasNumberOfCycles.kt

```diff
@@ -0,0 +1,58 @@
+package com.anaplan.engineering.azuki.graphs.analysis.directed
+
+import com.anaplan.engineering.azuki.core.runner.AnalysisScenario
+import com.anaplan.engineering.azuki.graphs.dsl.GraphScenario
+import com.anaplan.engineering.azuki.graphs.graphA
+
+class HasNumberOfCycles: GraphScenario() {
```

Comment:
:no_entry_sign: again, things like empty graph, edges to self, graphs with sub cycles (e.g. does it count as 1 or 2 if you have 1 shape with 2 cyclic paths)

some of these will be eacs, some should be analysis scenarios